src/openforms/contrib/objects_api/validators.py

```diff
@@ -78,3 +92,52 @@ def validate_document_type_references(
 
     if errors:
         raise ValidationError(errors)
+
+
+def validate_object_ownership(submission: Submission) -> None:
+    form = submission.form
+
+    try:
+        auth_info = submission.auth_info
+    except ObjectDoesNotExist:
+        raise PermissionDenied("Cannot pass data reference as anonymous user")
```

Comment:
what happens for a submission without data reference and no auth info?

Reply:
`plugin.verify_initial_data_ownership` is only called if `submission.initial_data_reference` is not empty (for both prefill and registration)

Reply:
Can you put in an explicit assert statement and docstring so that the expectations are made explicit?